[Bug]:

### Application

Outline Client

### Describe the bug

It seems like go 1.21 is required, but docs says it should be 1.20

### Steps to reproduce

docs

### What did you expect to happen?

all working

### What actually happened?

can't build android

### Outline Version

latest from github

### What operation system are you using?

Windows

### Operating System Version

dev 27686

### Screenshots and Videos

  ERROR(spawn_stream): go run github.com/go-task/task/v3/cmd/task -v client:tun2socks:android failed with exit code 1.}
    Printing stderr:
    C:\Users\etyso\go\pkg\mod\github.com\go-task\task\v3@v3.36.0\internal\omap\orderedmap.go:4:2: package cmp is not in GOROOT (C:\Program Files\Go\src\cmp)
    note: imported by a module that requires go 1.21
    C:\Users\etyso\go\pkg\mod\mvdan.cc\sh\v3@v3.8.0\interp\api.go:22:2: package maps is not in GOROOT (C:\Program Files\Go\src\maps)
    note: imported by a module that requires go 1.21
    C:\Users\etyso\go\pkg\mod\mvdan.cc\sh\v3@v3.8.0\expand\environ.go:8:2: package slices is not in GOROOT (C:\Program Files\Go\src\slices)
    note: imported by a module that requires go 1.21

    node:internal/process/esm_loader:40
          internalBinding('errors').triggerUncaughtException(
                                    ^
    C:\Users\etyso\go\pkg\mod\github.com\go-task\task\v3@v3.36.0\internal\omap\orderedmap.go:4:2: package cmp is not in GOROOT (C:\Program Files\Go\src\cmp)note: imported by a module that requires go 1.21C:\Users\etyso\go\pkg\mod\mvdan.cc\sh\v3@v3.8.0\interp\api.go:22:2: package maps is not in GOROOT (C:\Program Files\Go\src\maps)note: imported by a module that requires go 1.21C:\Users\etyso\go\pkg\mod\mvdan.cc\sh\v3@v3.8.0\expand\environ.go:8:2: package slices is not in GOROOT (C:\Program Files\Go\src\slices)note: imported by a module that requires go 1.21
    Thrown at:
        at loadESM (node:internal/process/esm_loader:40:33)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)


    Node.js v18.20.4
    ERROR(spawn_stream): node --trace-uncaught D:\outline\outline-apps\client\go\build.action.mjs android failed with exit code 1.}
    Printing stderr:
    C:\Users\etyso\go\pkg\mod\github.com\go-task\task\v3@v3.36.0\internal\omap\orderedmap.go:4:2: package cmp is not in GOROOT (C:\Program Files\Go\src\cmp)
    note: imported by a module that requires go 1.21
    C:\Users\etyso\go\pkg\mod\mvdan.cc\sh\v3@v3.8.0\interp\api.go:22:2: package maps is not in GOROOT (C:\Program Files\Go\src\maps)
    note: imported by a module that requires go 1.21
    C:\Users\etyso\go\pkg\mod\mvdan.cc\sh\v3@v3.8.0\expand\environ.go:8:2: package slices is not in GOROOT (C:\Program Files\Go\src\slices)
    note: imported by a module that requires go 1.21
    ERROR(spawn_stream): go run github.com/go-task/task/v3/cmd/task -v client:tun2socks:android failed with exit code 1.}
    Printing stderr:
    C:\Users\etyso\go\pkg\mod\github.com\go-task\task\v3@v3.36.0\internal\omap\orderedmap.go:4:2: package cmp is not in GOROOT (C:\Program Files\Go\src\cmp)
    note: imported by a module that requires go 1.21
    C:\Users\etyso\go\pkg\mod\mvdan.cc\sh\v3@v3.8.0\interp\api.go:22:2: package maps is not in GOROOT (C:\Program Files\Go\src\maps)
    note: imported by a module that requires go 1.21
    C:\Users\etyso\go\pkg\mod\mvdan.cc\sh\v3@v3.8.0\expand\environ.go:8:2: package slices is not in GOROOT (C:\Program Files\Go\src\slices)
    note: imported by a module that requires go 1.21

    node:internal/process/esm_loader:40
          internalBinding('errors').triggerUncaughtException(
                                    ^
    C:\Users\etyso\go\pkg\mod\github.com\go-task\task\v3@v3.36.0\internal\omap\orderedmap.go:4:2: package cmp is not in GOROOT (C:\Program Files\Go\src\cmp)note: imported by a module that requires go 1.21C:\Users\etyso\go\pkg\mod\mvdan.cc\sh\v3@v3.8.0\interp\api.go:22:2: package maps is not in GOROOT (C:\Program Files\Go\src\maps)note: imported by a module that requires go 1.21C:\Users\etyso\go\pkg\mod\mvdan.cc\sh\v3@v3.8.0\expand\environ.go:8:2: package slices is not in GOROOT (C:\Program Files\Go\src\slices)note: imported by a module that requires go 1.21
    Thrown at:
        at loadESM (node:internal/process/esm_loader:40:33)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)


    Node.js v18.20.4
  ▶ action(client/go/build): ❌ Failed.
  ERROR(spawn_stream): node --trace-uncaught D:\outline\outline-apps\client\src\cordova\build.action.mjs android failed with exit code 1.}
  Printing stderr:
    (node:10380) [DEP_WEBPACK_COMPILATION_ASSETS] DeprecationWarning: Compilation.assets will be frozen in future, all modifications are deprecated.
    BREAKING CHANGE: No more changes should happen to Compilation.assets after sealing the Compilation.
        Do changes to assets earlier, e. g. in Compilation.hooks.processAssets.
        Make sure to select an appropriate stage from Compilation.PROCESS_ASSETS_STAGE_*.
    (Use `node --trace-deprecation ...` to show where the warning was created)
    C:\Users\etyso\go\pkg\mod\github.com\go-task\task\v3@v3.36.0\internal\omap\orderedmap.go:4:2: package cmp is not in GOROOT (C:\Program Files\Go\src\cmp)
    note: imported by a module that requires go 1.21
    C:\Users\etyso\go\pkg\mod\mvdan.cc\sh\v3@v3.8.0\interp\api.go:22:2: package maps is not in GOROOT (C:\Program Files\Go\src\maps)
    note: imported by a module that requires go 1.21
    C:\Users\etyso\go\pkg\mod\mvdan.cc\sh\v3@v3.8.0\expand\environ.go:8:2: package slices is not in GOROOT (C:\Program Files\Go\src\slices)
    note: imported by a module that requires go 1.21
    ERROR(spawn_stream): go run github.com/go-task/task/v3/cmd/task -v client:tun2socks:android failed with exit code 1.}
    Printing stderr:
    C:\Users\etyso\go\pkg\mod\github.com\go-task\task\v3@v3.36.0\internal\omap\orderedmap.go:4:2: package cmp is not in GOROOT (C:\Program Files\Go\src\cmp)
    note: imported by a module that requires go 1.21
    C:\Users\etyso\go\pkg\mod\mvdan.cc\sh\v3@v3.8.0\interp\api.go:22:2: package maps is not in GOROOT (C:\Program Files\Go\src\maps)
    note: imported by a module that requires go 1.21
    C:\Users\etyso\go\pkg\mod\mvdan.cc\sh\v3@v3.8.0\expand\environ.go:8:2: package slices is not in GOROOT (C:\Program Files\Go\src\slices)
    note: imported by a module that requires go 1.21

    node:internal/process/esm_loader:40
          internalBinding('errors').triggerUncaughtException(
                                    ^
    C:\Users\etyso\go\pkg\mod\github.com\go-task\task\v3@v3.36.0\internal\omap\orderedmap.go:4:2: package cmp is not in GOROOT (C:\Program Files\Go\src\cmp)note: imported by a module that requires go 1.21C:\Users\etyso\go\pkg\mod\mvdan.cc\sh\v3@v3.8.0\interp\api.go:22:2: package maps is not in GOROOT (C:\Program Files\Go\src\maps)note: imported by a module that requires go 1.21C:\Users\etyso\go\pkg\mod\mvdan.cc\sh\v3@v3.8.0\expand\environ.go:8:2: package slices is not in GOROOT (C:\Program Files\Go\src\slices)note: imported by a module that requires go 1.21
    Thrown at:
        at loadESM (node:internal/process/esm_loader:40:33)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)


    Node.js v18.20.4
    ERROR(spawn_stream): node --trace-uncaught D:\outline\outline-apps\client\go\build.action.mjs android failed with exit code 1.}
    Printing stderr:
    C:\Users\etyso\go\pkg\mod\github.com\go-task\task\v3@v3.36.0\internal\omap\orderedmap.go:4:2: package cmp is not in GOROOT (C:\Program Files\Go\src\cmp)
    note: imported by a module that requires go 1.21
    C:\Users\etyso\go\pkg\mod\mvdan.cc\sh\v3@v3.8.0\interp\api.go:22:2: package maps is not in GOROOT (C:\Program Files\Go\src\maps)
    note: imported by a module that requires go 1.21
    C:\Users\etyso\go\pkg\mod\mvdan.cc\sh\v3@v3.8.0\expand\environ.go:8:2: package slices is not in GOROOT (C:\Program Files\Go\src\slices)
    note: imported by a module that requires go 1.21
    ERROR(spawn_stream): go run github.com/go-task/task/v3/cmd/task -v client:tun2socks:android failed with exit code 1.}
    Printing stderr:
    C:\Users\etyso\go\pkg\mod\github.com\go-task\task\v3@v3.36.0\internal\omap\orderedmap.go:4:2: package cmp is not in GOROOT (C:\Program Files\Go\src\cmp)
    note: imported by a module that requires go 1.21
    C:\Users\etyso\go\pkg\mod\mvdan.cc\sh\v3@v3.8.0\interp\api.go:22:2: package maps is not in GOROOT (C:\Program Files\Go\src\maps)
    note: imported by a module that requires go 1.21
    C:\Users\etyso\go\pkg\mod\mvdan.cc\sh\v3@v3.8.0\expand\environ.go:8:2: package slices is not in GOROOT (C:\Program Files\Go\src\slices)
    note: imported by a module that requires go 1.21

    node:internal/process/esm_loader:40
          internalBinding('errors').triggerUncaughtException(
                                    ^
    C:\Users\etyso\go\pkg\mod\github.com\go-task\task\v3@v3.36.0\internal\omap\orderedmap.go:4:2: package cmp is not in GOROOT (C:\Program Files\Go\src\cmp)note: imported by a module that requires go 1.21C:\Users\etyso\go\pkg\mod\mvdan.cc\sh\v3@v3.8.0\interp\api.go:22:2: package maps is not in GOROOT (C:\Program Files\Go\src\maps)note: imported by a module that requires go 1.21C:\Users\etyso\go\pkg\mod\mvdan.cc\sh\v3@v3.8.0\expand\environ.go:8:2: package slices is not in GOROOT (C:\Program Files\Go\src\slices)note: imported by a module that requires go 1.21
    Thrown at:
        at loadESM (node:internal/process/esm_loader:40:33)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)


    Node.js v18.20.4
  ▶ action(client/go/build): ❌ Failed.
▶ action(client/src/cordova/build): ❌ Failed.
PS D:\outline\outline-apps>

--- a/client/README.md
+++ b/client/README.md
@@ -9,7 +9,7 @@ The client's user interface is implemented in [Polymer](https://www.polymer-proj
 
 ## Requirements for all builds
 
-All builds require [Node](https://nodejs.org/) 18 (lts/hydrogen), and [Go](https://golang.org/) 1.20 installed in addition to other per-platform requirements.
+All builds require [Node](https://nodejs.org/) 18 (lts/hydrogen), and [Go](https://golang.org/) 1.21 installed in addition to other per-platform requirements.
 
 > 💡 NOTE: if you have `nvm` installed, run `nvm use` to switch to the correct node version!
 

